### ISSUE TITLE
Not passing right hostname to XMPP server

### ISSUE THREAD

**Issue by gloomytrousers** [matteocorti/check_ssl_cert#71]
I'm having trouble getting the plugin to validate my XMPP server's cert, because it doesn't pass the correct hostname via openssl:
```
[root@localhost ~]# /usr/lib64/nagios/plugins/check_ssl_cert  --host server.example.com --port 5222 --protocol xmpp --cn example.com -d
#...
[DBG] '/usr/bin/openssl s_client' supports '-xmpphost': using -xmpphost server.example.com
#...
[DBG] executing with timeout (15s): echo 'Q' | /usr/bin/openssl s_client   -starttls xmpp -connect server.example.com:5222 -xmpphost server.example.com -verify 6     2> /tmp/check_ssl_certky8Jdg 1> /tmp/check_ssl_certNtjjQp
[DBG]   /usr/bin/timeout 15 /bin/sh -c "echo 'Q' | /usr/bin/openssl s_client   -starttls xmpp -connect server.example.com:5222 -xmpphost server.example.com -verify 6     2> /tmp/check_ssl_certky8Jdg 1> /tmp/check_ssl_certNtjjQp"
[...]
Error: verify depth is 6
SSL_CERT CRITICAL server.example.com: No certificate returned
```
Running the underlying openssl command manually, there's no cert:
```
[root@localhost ~]# /usr/bin/openssl s_client   -starttls xmpp -connect server.example.com:5222 -xmpphost server.example.com -verify 6 
verify depth is 6
CONNECTED(00000003)
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 300 bytes and written 134 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
---
```
...but if I pass the right host name as -xmpphost, I get the expected result:
```
[root@localhost ~]# /usr/bin/openssl s_client   -starttls xmpp -connect server.example.com:5222 -xmpphost example.com -verify 6 
verify depth is 6
CONNECTED(00000003)
depth=2 O = Digital Signature Trust Co., CN = DST Root CA X3
verify return:1
depth=1 C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
verify return:1
depth=0 CN = example.com
verify return:1
---
Certificate chain
 0 s:/CN=example.com
   i:/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3
 1 s:/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3
   i:/O=Digital Signature Trust Co./CN=DST Root CA X3
#...
```
I can't see an obvious way to make the plugin pass the right name to the server. I tried using "--sni" but that doesn't work (maybe it should).

**Comment by matteocorti** [comment]
Hi, at the moment `-xmpphost` is set automatically. I will add an option similar to `-sni`

**Comment by gloomytrousers** [comment]
Not fixed I'm afraid. I installed 1.59 and passed `--xmpphost example.com` but the underlying `openssl` command still gets `-xmpphost server.example.com`.

**Comment by gloomytrousers** [comment]
Aha, got it: `XMPPHOST="-xmpphost ${HOST}"` on line 1074 should be `XMPPHOST="-xmpphost ${XMPPHOST:-$HOST}"` (this should use the value of `$HOST`, the original behaviour, if `--xmpphost` was not passed).

**Comment by matteocorti** [comment]
Sorry it was a typo. As I am working on macOS (and my OpenSSL does not have the `-xmpphost`) I was not able to test. Do you also have an example with a real server that I could include in the test suite?

**Comment by gloomytrousers** [comment]
Just picked one at random from https://list.jabber.at/, and testing with my locally-patched version...
```
[root@localhost ~]# /usr/lib64/nagios/plugins/check_ssl_cert   --host prosody.xmpp.is --port 5222 --protocol xmpp
SSL_CERT CRITICAL prosody.xmpp.is: No certificate returned
[root@localhost ~]# /usr/lib64/nagios/plugins/check_ssl_cert   --host prosody.xmpp.is --port 5222 --protocol xmpp --xmpphost xmpp.is
SSL_CERT OK - x509 certificate 'xmpp.is' from 'Let's Encrypt Authority X3' valid until Feb 15 20:23:18 2018 GMT (expires in 62 days)|days=62;;;;
```

**Comment by matteocorti** [comment]
Fixed in 1.60.0

**Comment by rodom** [comment]
Thanks, confirmed all fixed!
